components/clarinet-format/tests/golden-intended/clarity-bitcoin.clar

```diff
@@ -0,0 +1,1183 @@
+				}
+			}
+		)
+	)
+)
+
+;; Reads the next four bytes from txbuff as a little-endian 32-bit integer, and updates the index.
+;; Returns (ok { uint32: uint, ctx: { txbuff: (buff 4096), index: uint } }) on success.
+;; Returns (err ERR-OUT-OF-BOUNDS) if we read past the end of txbuff
+(define-read-only (read-uint32
+		(ctx {
+				txbuff: (buff 4096),
+				index: uint
+			}
+		)
+	)
+	(let (
+		(data (get txbuff ctx))
+		(base (get index ctx))
+	)
+		(ok
+			{
+				uint32: (buff-to-uint-le
+					(unwrap-panic
+						(as-max-len?
+							(unwrap! (slice? data base (+ base u4)) (err ERR-OUT-OF-BOUNDS)) u4
+						)
+					)
+				),
+				ctx: {
+					txbuff: data,
+					index: (+ u4 base)
+				}
+			}
+		)
+	)
+)
+
+;; Reads the next eight bytes from txbuff as a little-endian 64-bit integer, and updates the index.
+;; Returns (ok { uint64: uint, ctx: { txbuff: (buff 4096), index: uint } }) on success.
+;; Returns (err ERR-OUT-OF-BOUNDS) if we read past the end of txbuff
+(define-read-only (read-uint64
+		(ctx {
+				txbuff: (buff 4096),
+				index: uint
+			}
+		)
+	)
+	(let (
+		(data (get txbuff ctx))
+		(base (get index ctx))
+	)
+		(ok
+			{
+				uint64: (buff-to-uint-le
+					(unwrap-panic
+						(as-max-len?
+							(unwrap! (slice? data base (+ base u8)) (err ERR-OUT-OF-BOUNDS)) u8
+						)
+					)
+				),
+				ctx: {
+					txbuff: data,
+					index: (+ u8 base)
+				}
+			}
+		)
+	)
+)
+
+;; Reads the next varint from txbuff, and updates the index.
+;; Returns (ok { varint: uint, ctx: { txbuff: (buff 4096), index: uint } }) on success
+;; Returns (err ERR-OUT-OF-BOUNDS) if we read past the end of txbuff.
+(define-read-only (read-varint
+		(ctx {
+				txbuff: (buff 4096),
+				index: uint
+			}
+		)
+	)
+	(let (
+		(ptr (get index ctx))
+		(tx (get txbuff ctx))
+		(byte (buff-to-uint-le (unwrap! (element-at tx ptr) (err ERR-OUT-OF-BOUNDS))))
+	)
+		(if (<= byte u252)
+				;; given byte is the varint
+				(ok
+				{
+					varint: byte,
+					ctx: {
+						txbuff: tx,
+						index: (+ u1 ptr)
+					}
+				}
+			)
+				(if (is-eq byte u253)
+					(let (
+					;; next two bytes is the varint
+					(parsed-u16
+						(try!
+							(read-uint16 {
+									txbuff: tx,
+									index: (+ u1 ptr)
+								}
+							)
+						)
+					)
+				)
+					(ok
+						{
+							varint: (get uint16 parsed-u16),
+							ctx: (get ctx parsed-u16)
+						}
+					)
+				)
+					(if (is-eq byte u254)
+						(let (
+						;; next four bytes is the varint
+						(parsed-u32
+							(try!
+								(read-uint32
+									{
+										txbuff: tx,
+										index: (+ u1 ptr)
+									}
+								)
+							)
+						)
+					)
+						(ok
+							{
+								varint: (get uint32 parsed-u32),
+								ctx: (get ctx parsed-u32)
+							}
+						)
+					)
+						(let (
+						;; next eight bytes is the varint
+						(parsed-u64
+							(try!
+								(read-uint64
+									{
+										txbuff: tx,
+										index: (+ u1 ptr)
+									}
+								)
+							)
+						)
+					)
```

Comment:
What is the rule/logic for new lines here?
I feel like I'd be happy with something like
```clarity
            (let (
							;; next eight bytes is the varint
							(parsed-u64 (try! (read-uint64 {
								txbuff: tx,
								index: (+ u1 ptr)
							})))
						)
```

Reply:
It does the paren lining up here: https://github.com/hirosystems/clarinet/pull/1609/files#diff-bdfb6363ac810ef4d419caf59fba1e9b747a6a281fbd56b6d87a70cf66c226f4R779-R781

It's quite easy to stack the closing parens `)))` by not including the above logic. 

Maybe it makes sense in some situations, but this is generic inner content handling. It's also worsened by how bad tabs look here, I think..

Reply:
Beyond just stacking closing parens, I think we don't need to introcude new line before each opening one `(`
If the value was not a tuple (or was a variable name), it would easily fit on one line
```clarity
(parsed-u64 (try! (read-uint64 value)))
```

Here, I'd say we want to split the tuple into multiple lines for readability, but it doesn't imply splitting the rust of the line
```clarity
(parsed-u64 (try! (read-uint64 {
  txbuff: tx,
  index: (+ u1 ptr)
 })))
 ```
 
 
What do you think?